File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    zero = 0
    count = 0
    one = 0
    
    while not s == "1":
        zero += s.count('0')
        one = s.count('1')
        s = bin(one)[2:]
        count += 1
    
    return [count, zero]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int zero = 0;
        int num = 0;
        
        while (!s.equals("1")) {
            int one = 0;
            for (int i = 0; i < s.length(); i++) {
                if (s.charAt(i) == '0')         zero++;
                else if (s.charAt(i) == '1')    one++;
            }
            s = Integer.toBinaryString(one);
            num++;
        }
        return new int[] {num, zero};
    }
}